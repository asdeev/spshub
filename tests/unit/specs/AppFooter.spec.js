import { createLocalVue, shallowMount } from '@vue/test-utils';
import { expect } from 'chai';
import Router from 'vue-router';
import Vuetify from 'vuetify';
import testMap from '../../testMap';
import footer from './fixtures/footer';

const localVue = createLocalVue();
localVue.use(Router);
localVue.use(Vuetify);

describe('AppFooter', () => {
  let props;

  const build = () => {
    const wrapper = shallowMount(testMap.AppFooter, {
      localVue,
      propsData: props
    });

    return {
      wrapper,
      socialMedia: () => wrapper.find('.app-footer__social-media'),
      subheading: () => wrapper.find('.app-footer__subheading'),
      copyright: () => wrapper.find('.app-footer__copyright')
    };
  };

  beforeEach(() => {
    props = {
      footer
    };
  });

  it('renders the component', () => {
    const { wrapper } = build();

    expect(wrapper.html()).to.matchSnapshot();
  });

  it('renders the main components', () => {
    const { socialMedia, subheading, copyright } = build();

    expect(socialMedia().exists()).to.be.true;
    expect(subheading().exists()).to.be.true;

    expect(copyright().exists()).to.be.true;
  });
});
