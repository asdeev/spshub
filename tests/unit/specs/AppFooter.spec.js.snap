// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter renders the component 1`] = `"<v-footer-stub height=\\"auto\\"><v-card-stub tag=\\"div\\" tile=\\"true\\" flat=\\"true\\" class=\\"flex\\"><v-card-title-stub class=\\"app-footer__social-media primary2\\"><strong class=\\"app-footer__subheading subheading white--text\\">Get connected with us on social media!</strong> <v-spacer-stub></v-spacer-stub> <v-btn-stub activeclass=\\"v-btn--active\\" href=\\"link\\" tag=\\"button\\" target=\\"_blank\\" dark=\\"true\\" icon=\\"true\\" type=\\"button\\" class=\\"mx-3\\"><v-icon-stub size=\\"24\\">fab fa-facebook</v-icon-stub></v-btn-stub><v-btn-stub activeclass=\\"v-btn--active\\" href=\\"link\\" tag=\\"button\\" target=\\"_blank\\" dark=\\"true\\" icon=\\"true\\" type=\\"button\\" class=\\"mx-3\\"><v-icon-stub size=\\"24\\">fab fa-twitter</v-icon-stub></v-btn-stub><v-btn-stub activeclass=\\"v-btn--active\\" href=\\"link\\" tag=\\"button\\" target=\\"_blank\\" dark=\\"true\\" icon=\\"true\\" type=\\"button\\" class=\\"mx-3\\"><v-icon-stub size=\\"24\\">fab fa-instagram</v-icon-stub></v-btn-stub><v-btn-stub activeclass=\\"v-btn--active\\" href=\\"link\\" tag=\\"button\\" target=\\"_blank\\" dark=\\"true\\" icon=\\"true\\" type=\\"button\\" class=\\"mx-3\\"><v-icon-stub size=\\"24\\">fab fa-youtube</v-icon-stub></v-btn-stub></v-card-title-stub> <v-card-actions-stub class=\\"app-footer__copyright secondary white--text justify-end\\">  2019 </v-card-actions-stub></v-card-stub></v-footer-stub>"`;
